docs/en/middleware.rst

```diff
@@ -201,25 +201,98 @@ Both redirect handlers share the same configuration options:
+This excepction will always be an instance of ``Authorization\Exception\Exception``.
+In this example the ``Authorization.Redirect`` handler just gives you the option to 
+specify which exceptions you want to listen to.
+
+So in this example where we use the ``Authorization.Redirect`` handler we can
+add other ``Authorization\Exception\Exception`` based exceptions to the 
+``execeptions`` array if we wan't to handle them gracefully::
 
-You can also add your own handler. Handlers should implement
-``Authorization\Middleware\UnauthorizedHandler\HandlerInterface``, be suffixed
-with ``Handler`` suffix and reside under your app's or plugin's
-``Middleware\UnauthorizedHandler`` namespace.
+    'exceptions' => [
+        MissingIdentityException::class,
+        ForbiddenException::class
+    ],
+
+See https://github.com/cakephp/authorization/blob/master/src/Middleware/UnauthorizedHandler/RedirectHandler.php 
 
 Configuration options are passed to the handler's ``handle()`` method as the
 last parameter.
 
-Handlers catch only those exceptions which extend the
-``Authorization\Exception\Exception`` class.
+Add a flash message after being redirected by an unauthorized request
+------------------------------
+
+Currently there is no straightforward way to add a flash message to the unauthorized redirect.
+
+Therefore you need to create your own Handler which adds th flash message (or any 
+other logic you want to happen on redirect)
+
+How to create a custom UnauthorizedHandler
+------------------------------
+
+1) Create the folder ``src/Middleware/UnauthorizedHandler``
+2) Create a class with the namespace ``Middleware\UnauthorizedHandler`` which ends with the string ``Handler`` (like ``src/Middleware/UnauthorizedHandler/CustomRedirectHandler.php``)::
+
+    <?php
+    declare(strict_types=1);
+    namespace App\Middleware\UnauthorizedHandler;
+    
+    class CustomRedirectHandler {...}
+
+3) If you want to have the basic redirect logic then extend your class with::
+
+    class CustomRedirectHandler extends \Authorization\Middleware\UnauthorizedHandler\RedirectHandler
+
+4) If not add the ``Authorization\Middleware\UnauthorizedHandler\HandlerInterface`` to your class::
+
+    class CustomRedirectHandler implements \Authorization\Middleware\UnauthorizedHandler\HandlerInterface
+
+5) Add the flash message logic inside the ``handle()`` method like so::
+
+    use Authorization\Exception\Exception;
+    use Psr\Http\Message\ServerRequestInterface;
+    use Psr\Http\Message\ResponseInterface;
+
+    public function handle(Exception $exception, ServerRequestInterface $request, array $options = []): ResponseInterface {
+        $response = parent::handle($exception, $request, $options);
+        $request->getFlash()->error('You are not authorized to access that location');
+        return $response;
+    }
+
+6) After you are done with the implementation of that function tell the middleware you want to use your custom handler via::
```

Comment:
```suggestion
#. After you are done with the implementation of that function tell the middleware you want to use your custom handler via::
```

Reply:
Instead of a 6 step process, you could give the entire class as a single code block and instruct the reader where to save it. That is how examples work elsewhere in the book. Readers often have a hard time figuring out how to assemble a class from small snippets.

Reply:
Yea, good idea. I will do that

Reply:
I have now reduced the example to a 2 step process. I hope this is now more clear and easier to understand